weave/upload.py

```diff
@@ -66,16 +67,15 @@ def derive_integrity_data(file_path, byte_count=10**8):
      {
       'file_size': bytes (int),
       'hash': sha256 hash (string),
-      'access_date': current date/time (string)
-      'source_path': path to the original source of data (string)
+      'access_date': current date/time (string),
+      'source_path': path to the original source of data (string),
       'byte_count': byte count used for generated checksum (int)
      }
     """
-
+    source_file_system = kwargs.get("source_file_system",LocalFileSystem())
```

Comment:
I like how initialized this. Is there any validation done elsewhere to ensure that this `source_file_system` is of a correct type? Is there a base class that it would derive from that would have all desired methods (i.e. `AsyncFileSystem`)?